.github/workflows/release.yml

```diff
@@ -0,0 +1,111 @@
+
+    - name: Install Golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+
+    - name: Build Image
+      run: |
+        TAG=$GITHUB_REF_NAME
+        docker build -t goharbor/harbor-acceld:$TAG -f script/release/Dockerfile .
+
+    - name: Test Image
+      # Enable tty for docker
+      shell: 'script -q -e -c "bash {0}"'
+      run: |
+        TAG=$GITHUB_REF_NAME
+        docker run -v $PWD/misc/config/config.yaml.nydus.tmpl:/etc/acceld-config.yaml -it -d --rm -p 2077:2077 goharbor/harbor-acceld:$TAG /etc/acceld-config.yaml
+        sleep 5
+        curl -f http://127.0.0.1:2077/api/v1/health
+
+    - name: Login Registry
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+    - name: Push Image
+      run: |
+        TAG=$GITHUB_REF_NAME
+        # Only push for non pull_request
+        if [ "$GITHUB_EVENT_NAME" != "pull_request" ]; then
+          docker push goharbor/harbor-acceld:$TAG
+        fi
+
+  release_binary:
+    runs-on: ubuntu-latest
+    env:
+      GO_VERSION: 1.17
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Cache Go Mod
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go
+      - name: Build Binary
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.43.0
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: harbor-acceld-artifacts
+          path: |
+            acceld
+            accelctl
+
+  upload_binary:
+    runs-on: ubuntu-latest
+    needs: [release_binary]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Hub
+        run: |
+          HUB_VER=$(curl -s "https://api.github.com/repos/github/hub/releases/latest" | jq -r .tag_name | sed 's/^v//')
+          wget -q -O- https://github.com/github/hub/releases/download/v$HUB_VER/hub-linux-amd64-$HUB_VER.tgz | \
+          tar xz --strip-components=2 --wildcards '*/bin/hub'
+          sudo mv hub /usr/local/bin/hub
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: harbor-acceld-artifacts
+          path: harbor-acceld
+      - name: Upload Artifacts
+        run: |
+          tag=$(echo $GITHUB_REF | cut -d/ -f3-)
+          tarball="harbor-acceld-$tag-x86_64.tgz"
+          chmod +x harbor-acceld/*
+          tar cf - harbor-acceld | gzip > ${tarball}
+          echo "tag=$tag" >> $GITHUB_ENV
+          echo "tarball=$tarball" >> $GITHUB_ENV
```

Comment:
Can we also upload the sha256sum of the release artifacts?

Reply:
Fixed